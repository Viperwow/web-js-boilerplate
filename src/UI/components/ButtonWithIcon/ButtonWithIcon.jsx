import {identity as _identity} from 'lodash';
import React from 'react';
import Button from 'src/UI/components/Button/Button';
import EyeIcon from 'assets/img/eye.svg';

const ButtonWithIcon = ({
  text,
  t = _identity,
}) => (
  <div className="flex flex-row items-center">
    <EyeIcon className="mr-2" />
    <Button t={t} text={t(text)} />
  </div>
);

export default ButtonWithIcon;
