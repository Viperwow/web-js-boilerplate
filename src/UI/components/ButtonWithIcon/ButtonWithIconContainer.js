import withTranslate from 'src/common/withTranslate';
import connectQuery from 'src/common/connectQuery';
import {compose} from 'src/helpers/utility';
import {GET_APP_LOCALE} from 'src/queries';
import ButtonWithIcon from './ButtonWithIcon';

const mapStateToProps = () => ({
  text: 'components.button',
});

export default compose(
  withTranslate,
  connectQuery(GET_APP_LOCALE, mapStateToProps),
)(ButtonWithIcon);
