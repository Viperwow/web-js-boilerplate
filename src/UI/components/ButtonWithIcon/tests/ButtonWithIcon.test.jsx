import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import ButtonWithIcon from '../ButtonWithIcon';

jest.unmock('../ButtonWithIcon');

describe('ButtonWithIcon component should render correctly', () => {
  test('With default props', () => {
    const output = shallow(<ButtonWithIcon />);

    expect(shallowToJson(output))
      .toMatchSnapshot();
  });

  test('With text prop', () => {
    const output = shallow(<ButtonWithIcon text="text" />);

    expect(shallowToJson(output))
      .toMatchSnapshot();
  });
});
