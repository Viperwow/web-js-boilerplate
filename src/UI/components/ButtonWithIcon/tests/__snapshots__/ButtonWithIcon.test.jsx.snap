// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonWithIcon component should render correctly With default props 1`] = `
<div
  className="flex flex-row items-center"
>
  <SvgrURL
    className="mr-2"
  />
  <Button
    t={[Function]}
  />
</div>
`;

exports[`ButtonWithIcon component should render correctly With text prop 1`] = `
<div
  className="flex flex-row items-center"
>
  <SvgrURL
    className="mr-2"
  />
  <Button
    t={[Function]}
    text="text"
  />
</div>
`;
